src/mapping-pmp-v0.ts

```diff
@@ -0,0 +1,244 @@
+
+      // load or create the latest state entity for this PMP
+      const latestStateId = generateLatestStateId(
+        event.address,
+        event.params.tokenId,
+        pmpInput.key
+      );
+      let latestState = PMPLatestState.load(latestStateId);
+
+      if (!latestState) {
+        latestState = new PMPLatestState(latestStateId);
+      } else {
+        latestNonce = latestState.latestNonce.plus(BigInt.fromI32(1));
+      }
+
+      const pmpId = generatePMPId(
+        event.address,
+        event.params.tokenId,
+        pmpInput.key,
+        latestNonce
+      );
+
+      let pmp = new PMP(pmpId);
+      pmp.key = pmpInput.key;
+      pmp.token = token.id;
+
+      let currentConfiguredParamType = getPMPParamTypeString(
+        BigInt.fromI32(pmpInput.configuredParamType)
+      );
+
+      pmp.tokenPMPNonce = latestNonce;
+      pmp.configuredParamType = currentConfiguredParamType;
+      pmp.configuredValue = pmpInput.configuredValue.toHexString();
+
+      if (pmpInput.configuringArtistString) {
+        pmp.artistConfiguredValueString = pmpInput.configuredValueString;
+      } else {
+        pmp.nonArtistConfiguredValueString = pmpInput.configuredValueString;
+      }
+
+      // update the token pmp
+      pmp.updatedAt = event.block.timestamp;
+      pmp.save();
+
+      // update the latest state entity
+      latestState.latestNonce = latestNonce;
+      latestState.save();
+    }
+  }
+}
+
+export function handleProjectConfigured(event: ProjectConfigured): void {
+  let projectId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.projectId
+  );
+  let project = Project.load(projectId);
+
+  if (project) {
+    const projectConfigId = generateProjectConfigId(
+      event.address,
+      event.params.projectId
+    );
+
+    let projectConfig = PMPProjectConfig.load(projectConfigId);
+    if (!projectConfig) {
+      projectConfig = new PMPProjectConfig(projectConfigId);
+      projectConfig.project = project.id;
+      projectConfig.save();
+    } else {
+      // clean up old pmp config, only store the most recent configurations
+      let previousPMPConfigCount = projectConfig.pmpConfigCount;
+      let storedKeys = projectConfig.pmpConfigKeys;
+
+      if (previousPMPConfigCount.toI32() != storedKeys.length) {
+        log.warning(
+          "Inconsistent data found for project config {}: count is {} but keys array length is {}",
+          [
+            projectConfigId,
+            previousPMPConfigCount.toString(),
+            storedKeys.length.toString()
+          ]
+        );
+      } else if (previousPMPConfigCount.gt(BigInt.fromI32(0))) {
+        // delete all old pmp configs
+        for (let i = 0; i < previousPMPConfigCount.toI32(); i++) {
+          const prevPMPConfigId = generatePMPConfigId(
+            event.address,
+            event.params.projectId,
+            storedKeys[i]
+          );
+          store.remove("PMPConfig", prevPMPConfigId);
+        }
```

Comment:
I think it's fine to only store the most recent PMP configs, because that is all that is ever used.
I was thinking if we should add `projectConfigNonce` to the `PMPProjectConfig` entity, then just store all PMP configs, relying on a `where` clause in graphql query to only get the latest PMP configs, but cleaning up and storing only most recent PMP configs seems most useful. I think that means there is an argument for not indexing `highestConfigNonce` on PMPConfig though...

Reply:
Agreed - I couldn't think of a use case for storing project config revisions.. Removing highestConfigNonce from PMPConfig makes sense, also reduces complexity a bit - I removed that from the entity and updated `updatedAt` to `createdAt` on the PMPConfig entity 🙏